## Basic Problem
Python merupakan salah satu dari bahasa pemrograman yang saat ini sangat popular digunakan. Python menyediakan banyak library dan method yang dapat dimanfaatkan untuk membuat program, memproses data, maupun analisis data. Salah satu yang dapat digunakan dalam menganalisis data yaitu Pandas. Pandas merupakan library yang dibangun diatas Numpy yang berfungsi untuk melakukan preprocessing dan analisis data. Library Pandas dapat mengolah data yang berbentuk csv, txt, excel, html, dan sebagainya.


Dimisalkan saya memiliki suatu dataset food delivery order. Dari dataset ini saya ingin mengetahui 5 harga makanan terendah, fee delivery termahal dan jumlah orderan yang cancelled dan complete ada berapa. Pada kesempatan kali ini saya akan mencoba melakukan sorting, filtering, dan grouping Data Frame untuk dapat menyelesaikan permasalahan tersebut. Disini data yang akan digunakan adalah data Food Order Delivery. 

## 1. *Sorting*

Hal pertama yang kita lakukan ialah mengimport library nya terlebih dahulu

In [1]:
import numpy as np
import pandas as pd

kemudian kita baca data CSV yang telah kita sediakan

In [7]:
food = pd.read_csv('food_delivery_datasets.csv')
food

order_id  voucher_amount  ...     status  random_id
0     FOOD-10455               0  ...  Completed   0.511149
1     FOOD-10985               0  ...  Completed   0.832936
2     FOOD-10911               0  ...  Completed   0.828026
3     FOOD-10932               0  ...  Completed   0.239377
4     FOOD-10961               0  ...  Completed   0.995587
...          ...             ...  ...        ...        ...
1071  FOOD-10921               0  ...  Completed   0.248328
1072  FOOD-10265            5000  ...  Completed   0.268251
1073  FOOD-10534               0  ...  Completed   0.132528
1074  FOOD-10158               0  ...  Completed   0.167946
1075  FOOD-10770               0  ...  Completed   0.358051

[1076 rows x 11 columns]

setelah itu kita akan mencoba melakukan sorting data secara ascending dan descending untuk dapat melihat urutan data dari yang terendah dan tertinggi.  Kali ini akan dilakukan sorting food_price Dataframe Food_Delivery secara ascending agar dapat mengetahui urutan harga makanan dari yang terendah hingga tertinggi

In [8]:
data1 = food.sort_values(by='food_price', ascending=True)
data1

order_id  voucher_amount  ...     status  random_id
202   FOOD-10654               0  ...  Completed   0.716985
412   FOOD-10898               0  ...  Completed   0.751653
858   FOOD-10395               0  ...  Completed   0.093072
851   FOOD-11013               0  ...  Completed   0.753612
864   FOOD-10439               0  ...  Completed   0.815192
...          ...             ...  ...        ...        ...
1037  FOOD-10781               0  ...  Completed   0.674128
827   FOOD-10565               0  ...  Completed   0.912690
67    FOOD-10633               0  ...  Completed   0.734508
10    FOOD-10386               0  ...  Completed   0.485939
128   FOOD-10212               0  ...  Completed   0.469235

[1076 rows x 11 columns]

selanjutnya akan dilakukan sorting delivery_fee Dataframe Food secara descending agar dapat mengetahui urutan delivery_fee dari yang tertinggi hingga terendah. Dalam sorting descending dapat ditambahkan parameter ascending=false

In [10]:
data2 = food.sort_values(by='delivery_fee', ascending=False)
data2

order_id  voucher_amount  ...     status  random_id
1032  FOOD-10382               0  ...  Completed   0.138681
374   FOOD-10845               0  ...  Cancelled   0.304505
731   FOOD-10381               0  ...  Cancelled   0.894978
257   FOOD-10460               0  ...  Completed   0.587890
373   FOOD-10380               0  ...  Cancelled   0.837287
...          ...             ...  ...        ...        ...
627   FOOD-10936            5000  ...  Cancelled   0.256656
607   FOOD-10766            4000  ...  Completed   0.770513
377   FOOD-11019            2000  ...  Completed   0.403035
378   FOOD-10498            5000  ...  Cancelled   0.438224
538   FOOD-10219            5000  ...  Completed   0.469536

[1076 rows x 11 columns]

## 2.*Filtering* 
Dari hasil sortingan diatas kita akan mencoba melakukan filtering untuk menemukan  5 harga makanan terendah dan delivery fee termahal . 
selanjutnya kita ambil data dari dataframe menggunakan rentang kolom dan baris. untuk menemukan  5 harga makanan terendah kita tampilkan data dari dataframe baris 1 sampai 5 dari hasil sorting food_price Dataframe Food_Delivery secara ascending


In [12]:
data1[0:5]

order_id  voucher_amount  ...     status  random_id
202  FOOD-10654               0  ...  Completed   0.716985
412  FOOD-10898               0  ...  Completed   0.751653
858  FOOD-10395               0  ...  Completed   0.093072
851  FOOD-11013               0  ...  Completed   0.753612
864  FOOD-10439               0  ...  Completed   0.815192

[5 rows x 11 columns]

Untuk menemukan delivery fee termahal kita tampilkan data dari dataframe baris 0 sampai 1 dari hasil sorting delivery_fee Dataframe Food_Delivery secara descending 

In [13]:
data2[0:1]

order_id  voucher_amount  ...     status  random_id
1032  FOOD-10382               0  ...  Completed   0.138681

[1 rows x 11 columns]

kemudian kita pilih satu kolom pada Dataframe untuk melihat kolom status saja yang ditampilkan, kita dapat menggunakan []

In [15]:
status_order = food["status"] 
status_order

0       Completed
1       Completed
2       Completed
3       Completed
4       Completed
          ...    
1071    Completed
1072    Completed
1073    Completed
1074    Completed
1075    Completed
Name: status, Length: 1076, dtype: object

## 3.*Grouping*
Dalam pandas kita juga dapat melakukan pengelompokkan data berdasarkan nama tertentu dan mendapatkan nilainya, entah itu jumlahnya, rata-rata atau perhitungan totalnya. pada dataset Food_Delivery kita dapat melihat  jumlah orderan yang cancelled dan completed 

In [17]:
pd.DataFrame(food.groupby(['status'])['status'].size())

status
status           
Cancelled     195
Completed     881

## Kesimpulan
Berdasarkan hasil sorting, filtering, dan grouping, dapat diketahui bahwa


1.   5 harga terendah dari makanan yang telah diorder ialah 12900, 14200, 15200, 15700, 16000,
2.   fee delivery termahal ialah 44000
3.   jumlah orderan yang cancelled sebanyak 195 dan jumlah orderan yang completed sebanyak 881



 

